### PR TITLE
refactor(netflix): convert tableau module to TS

### PR DESCRIPTION
**Pull request by icfantv**
* convert tableau module to typescript.

### CODE REVIEW

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {IComponentController, ISCEService, module} from 'angular';
+
+import {AUTHENTICATION_SERVICE, AuthenticationService} from 'core/authentication/authentication.service';
+import {NetflixSettings} from '../../netflix.settings';
+import {Application} from 'core/application/application.model';
+
+require('../tableau.less');
```
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: import

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  static get $inject(): string[] {
+    return ['$sce', 'app', 'authenticationService'];
+  }
+
+  constructor(private $sce: ISCEService,
+              private app: Application,
+              private authenticationService: AuthenticationService) {
+
+    const user: string[] = this.authenticationService.getAuthenticatedUser().name.split('@');
+    const url =
```
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: This wrapping is super confusing, do you have an IDE setting?

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      NetflixSettings.tableau.appSourceUrl.replace('${app}', this.app.name).replace('${user}', user[0]);
```
Comment (icfantv) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: So this is the only thing that's _really_ different.  The original file did `[foo] = 'bar@bat.com'.split('@');` which results in `foo === 'bar'` but I don't know what it's called to look it up.  Happy to change `user[0]` back as it definitely feels weird.

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    this.srcUrl = this.$sce.trustAsResourceUrl(url);
+  }
+}
+
+export const TABLEAU_APPLICATION_CONTROLLER = 'spinnaker.netflix.tableau.application.controller';
+module(TABLEAU_APPLICATION_CONTROLLER, [AUTHENTICATION_SERVICE])
+  .controller('AppTableauCtrl', ApplicationTableauController);
```
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: Keep name ordering the same; either it is `TABLEAU_APPLICATION` and `TableauApp` or the inverse. Probably the inverse.

File: app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import {ISCEService, module} from 'angular';
+
+import {AUTHENTICATION_SERVICE, AuthenticationService} from 'core/authentication/authentication.service';
+import {NetflixSettings} from '../../netflix.settings';
+
+require('../tableau.less');
```
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts]: Import

File: app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(private $sce: ISCEService,
+              private authenticationService: AuthenticationService) {
+    const user: string[] = this.authenticationService.getAuthenticatedUser().name.split('@');
+    const url: string = NetflixSettings.tableau.summarySourceUrl.replace('${user}', user[0]);
+    this.srcUrl = this.$sce.trustAsResourceUrl(url);
+  }
+}
+
+export const TABLEAU_SUMMARY_CONTROLLER = 'spinnaker.netflix.tableau.summary.controller';
+module(TABLEAU_SUMMARY_CONTROLLER, [AUTHENTICATION_SERVICE])
```
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts]: Same string ordering as above.

File: app/scripts/modules/netflix/tableau/tableau.dataSource.ts [hunk 7]
```diff
@@ -1,14 +1,15 @@
+import {module} from 'angular';
+
 import {DataSourceConfig} from 'core/application/service/applicationDataSource';
-import {APPLICATION_DATA_SOURCE_REGISTRY} from 'core/application/service/applicationDataSource.registry';
+import {
+  APPLICATION_DATA_SOURCE_REGISTRY,
+  ApplicationDataSourceRegistry
+} from 'core/application/service/applicationDataSource.registry';
```
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/tableau.dataSource.ts]: Wrapping

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts (hunk 3 above)
Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: Array destructuring. I don't mind either way.

**Comment by icfantv** [comment]
@jrsquared sorry, this wasn't quite ready for review.  I was reviewing the PR when Milo needed to eat (and when baby's gotta eat, baby's gotta eat).  I'll make the changes and then ping ya'll.

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts (hunk 1 above)
Comment (icfantv) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: Didn't know we could do that w/ LESS files.  GTK and I like it way more betterer.

File: app/scripts/modules/netflix/tableau/application/appTableau.controller.ts (hunk 2 above)
Comment (icfantv) [inline app/scripts/modules/netflix/tableau/application/appTableau.controller.ts]: uhhhhh....YES.

File: app/scripts/modules/netflix/tableau/tableau.dataSource.ts (hunk 7 above)
Comment (icfantv) [inline app/scripts/modules/netflix/tableau/tableau.dataSource.ts]: So this wrapping I actually like.  When you import too many things from the same file (or a few _long_ things) it gets impossible to read.  Did you guys decide what to do in this case?

**Comment by icfantv** [comment]
@jrsquared @anotherchrisberry @christopherthielen PTAL (updated w/ JRs requests)

**Comment by anotherchrisberry** [comment]
LGTM

**Comment by jrsquared** [comment]
According to the [patch docs](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch), all of these commits changing to TS should probably be `refactor(...` instead of `chore(...`

File: app/scripts/modules/netflix/tableau/tableau.dataSource.ts (hunk 7 above)
Comment (christopherthielen) [inline app/scripts/modules/netflix/tableau/tableau.dataSource.ts]: My 2 cents: on collaborative projects, I prefer choosing a "rule" that the linter can enforce. 

I don't much care what that rule is, but hopefully it's one that IntelliJ can apply using `cmd-option-O` (optimize imports)

IntelliJ offers 2 choices: "wrap if long" and "chop down if long" (I personally prefer "wrap if long")

File: app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import {NetflixSettings} from '../../netflix.settings';
+
+import '../tableau.less';
+
+class SummaryTableauController {
+
+  public srcUrl: string;
+
+  static get $inject(): string[] {
+    return ['$sce', 'app', 'authenticationService'];
```
Comment (christopherthielen) [inline app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts]: This inject doesn't match the constructor below. Looks like a bug

Comment (jrsquared) [inline app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts]: Good catch!

Comment (anotherchrisberry) [inline app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts]: @icfantv how did you test this?

Comment (icfantv) [inline app/scripts/modules/netflix/tableau/summary/summaryTableau.controller.ts]: I clicked the analytics tab, waited for all the UI graphs to load, and confirmed that the URL requested in the network stack matched the one requested by the controller.  Not sure how it got changed between testing and committing.  Apologies.  Thanks @christopherthielen for catching and @jrsquared for fixing.

**Pull request merged**